lib/patch/listener.ex

```diff
@@ -0,0 +1,111 @@
+defmodule Patch.Listener do
+  use GenServer
+
+  @default_timeout 5000
+
+  @typedoc """
+  Option to control how long the listener should wait for GenServer.call
+
+  Value is either the number of milliseconds to wait or the `:infinity` atom.
+
+  Defaults to #{@default_timeout}
+  """
+  @type timeout_option :: {:timeout, timeout()}
+
+  @typedoc """
+  Sum-type of all valid options
+  """
+  @type option :: timeout_option()
+
+  @typedoc """
+  Convenience type for list of options
+  """
+  @type options :: [option()]
+
+  @type t :: %__MODULE__{
+          recipient: pid(),
+          tag: atom(),
+          target: pid(),
+          timeout: timeout()
+        }
+  defstruct [:recipient, :tag, :target, :timeout]
+
+  ## Client
+
+  def child_spec(args) do
+    recipient = Keyword.fetch!(args, :recipient)
+    tag = Keyword.fetch!(args, :tag)
+    target = Keyword.fetch!(args, :target)
+    options = Keyword.get(args, :options, [])
+
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [recipient, tag, target, options]}
+    }
+  end
+
+  @spec start_link(recipient :: atom(), tag :: atom(), target :: pid() | atom(), options()) ::
+          {:ok, pid()} | {:error, :not_found}
+  def start_link(recipient, tag, target, options \\ [])
+
+  def start_link(recipient, tag, target, options) when is_atom(target) do
+    case Process.whereis(target) do
+      nil ->
+        {:error, :not_found}
+
+      pid ->
+        true = Process.unregister(target)
```

Comment:
This makes a lot of senses and is a straightforward way to do it. However if I understand correctly it does mean we must always run our tests serially right? maybe worth a caveat in the README, or ... is it possible to enforce?

Reply:
The entire library is dependent on mocking global state so async is out of the question, but the README does not mention this.  I'll add an admonition.

Reply:
In the README there is now a `Limitations` section that details that this library is not compatible with `async: true`